internal/runners/update/update.go

```diff
@@ -40,75 +38,14 @@ func New(prime primeable) *Update {
 }
 
 func (u *Update) Run(params *Params) error {
-	return run(u.prompt, params.Lock, u.isLocked(), params.Force, u.runLock, u.runUpdateLock, u.runUpdateGlobal, confirmUpdateLock)
-}
-
-func run(prompt prompt.Prompter, lock, isLocked, force bool, runLock, runUpdateLock, runUpdateGlobal func() error, confirmLock func(prompt.Prompter) error) error {
-	if lock {
-		return runLock()
-	}
-	if !lock && isLocked {
-		if !force {
-			if err := confirmLock(prompt); err != nil {
-				return locale.WrapError(err, "err_update_lock_confirm", "Could not confirm whether to update.")
-			}
-		}
-		return runUpdateLock()
-	}
-	return runUpdateGlobal()
-}
-
-func (u *Update) runLock() error {
-	u.out.Notice(locale.Tl("locking_version", "Locking State Tool to the current version."))
-
-	if u.project.Lock() != "" {
-		u.out.Print(locale.Tl("lock_project_uptodate", "Your project is already locked, did you mean to run 'state update' (without the --lock flag)?"))
-		return nil
-	}
-
-	err := projectfile.AddLockInfo(u.project.Source().Path(), constants.BranchName, constants.Version)
-	if err != nil {
-		return locale.WrapError(err, "err_update_projectfile", "Could not update projectfile")
-	}
-
-	u.out.Print(locale.Tl("version_locked", "Version locked at {{.V0}}", constants.Version))
-	return nil
-}
-
-func (u *Update) runUpdateLock() error {
-	u.out.Notice(locale.Tl("updating_lock_version", "Locking State Tool to latest version available for your project."))
-
-	info, err := updater.New(u.project.Version()).Info()
-	if err != nil {
-		return locale.WrapError(err, "err_update_updater", "Could not retrieve update information.")
-	}
-
-	if info == nil {
-		u.out.Print(locale.Tl("update_project_uptodate", "Your project is already using the latest State Tool version available."))
-		return nil
-	}
-
-	err = projectfile.AddLockInfo(u.project.Source().Path(), info.Version, constants.BranchName)
-	if err != nil {
-		return locale.WrapError(err, "err_update_projectfile", "Could not replace update in projectfile")
-	}
-
-	u.out.Print(locale.Tl("version_lock_updated", "Locked version updated to {{.V0}}", constants.Version))
-	return nil
-}
-
-func (u *Update) runUpdateGlobal() error {
 	u.out.Notice(locale.Tl("updating_version", "Updating State Tool to latest version available."))
 
+	channel := fetchChannel(params.Channel, true)
```

Comment:
So, here are ignoring the channel name configured in the activestate.yaml file, and instead just update or switch the "global" channel of the installed state tool. That confused me for a while, but it makes sense!  I actually like this separation! :+1: